vassal-app/src/main/java/VASSAL/build/module/BasicCommandEncoder.java

```diff
@@ -151,30 +195,42 @@
     Map.entry(GlobalHotKey.ID, GlobalHotKey::new)
   );
 
+  /**
+   * Fallthrough factory to catch unknown types.
+   */
   private final DecoratorFactory defaultDecoratorFactory = (type, inner) -> {
-    System.err.println("Unknown type " + type);
+    ErrorDialog.dataWarning(new BadDataReport("Unknown type " + type + " not found in BasicCommandEncoder's list of traits and basic pieces.", ""));
     return new Marker(Marker.ID, inner);
   };
 
+  /**
+   * Why do I exist? No one knows!
+   */
   public BasicCommandEncoder() {
   }
```

Comment:
An empty nullary ctor is generated automatically if there is no other ctor defined, so this could be removed entirely.

Reply:
Was sort of my guess, will do.